src/merge.py

```diff
@@ -15,8 +18,21 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     z = []  # a new list to copy elements into
     # FIXME: fill out the loop so you merge the lists
     # until one of them is empty
+
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
+        if x[i] <= y[j]: 
+            z.append(x[i])
+            i += 1
+        if y[j] <= x[i]: 
+            z.append(y[j])
+            j += 1
+        if i < len(x) and j == len(y):
+            z.append(x[i])
+        if j < len(y) and i == len(x):
+            z.append(x[j])
```

Comment:
There is a slight problem here. If one of the lists is not completely processed, there might be more than one element to copy. You check if, for example, `x` is not completely done (`i < len(x)`) and then you append the next element in `x`, `x[i]`. But imagine that we started with

```python
x = [1, 2, 3]
y = []
```

We leave the loop right away since `j == len(y)` and when we leave the loop `i == 0`. We observe that `i < len(x)` so you add `x[i] = x[0] = 1` to `z`, so that list is now `z == [1]`. But, if you return that, you get the wrong answer. The right merge of `[1, 2, 3]` and `[]` should be `[1, 2, 3]`.

Instead, you want to copy the entirety of the missing elements, and you can do that by slicing from `i` to the end, `x[i:]` for `x`, or from `j` to the end, `y[j:]`, depending on which list has missing elements. You can do that with a loop where you `append`, but there is already a method for copying a sequence of elements, `extend`. That would give you this code:

```python
        if i < len(x) and j == len(y):
             z.extend(x[i])
         if j < len(y) and i == len(x):
             z.extend(x[j])
```

But you can simplify it from a couple of observations. First, we already know that one of the lists are completely done, so we don't need to check the `j == len(y)` or `i == len(x)`. Those are implicitly true if we know that the other list isn't completely done. That gives us this code:

```python
        if i < len(x):
             z.extend(x[i:])
         if j < len(y):
             z.extend(x[j:])
```

But then we can ask ourselves what happens if we call `z.extend()` with an empty list? The only thing the `if` statements are checking is which list has more elements, but if `i == len(i)`, for example, then `x[i:]` is just the empty list, and is there any problem with copying that? It turns out that there isn't, so we don't need the `if`-statements at all. We can simply do:

```python
        z.extend(x[i:])
        z.extend(x[j:])
```